plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMetadataFileOperations.java

```diff
@@ -75,90 +83,220 @@ protected DistributedQueryRunner createQueryRunner()
+                        .addCopies(new FileOperation(METADATA, INPUT_FILE_NEW_STREAM), 1)
+                        .build());
     }
 
-    private static class OperationCounts
+    @Test
+    public void testShowStatsForTable()
+            throws Exception
     {
-        private final Map<OperationContext, Integer> allCounts;
-        private Optional<String> queryId = Optional.empty();
-        private Predicate<String> pathPredicate = path -> true;
-        private Optional<TrackingFileIoProvider.OperationType> type = Optional.empty();
+        assertUpdate("CREATE TABLE test_show_stats AS SELECT 2 as age", 1);
+
+        assertFileSystemAccesses("SHOW STATS FOR test_show_stats",
+                ImmutableMultiset.builder()
+                        .addCopies(new FileOperation(SNAPSHOT, INPUT_FILE_GET_LENGTH), 1)
+                        .addCopies(new FileOperation(SNAPSHOT, INPUT_FILE_NEW_STREAM), 1)
+                        .addCopies(new FileOperation(METADATA, INPUT_FILE_NEW_STREAM), 1)
+                        .addCopies(new FileOperation(MANIFEST, INPUT_FILE_GET_LENGTH), 2)
+                        .addCopies(new FileOperation(MANIFEST, INPUT_FILE_NEW_STREAM), 2)
+                        .build());
+    }
 
-        public OperationCounts(Map<OperationContext, Integer> allCounts)
+    @Test
+    public void testShowStatsForTableWithFilter()
+            throws Exception
+    {
+        assertUpdate("CREATE TABLE test_show_stats_with_filter AS SELECT 2 as age", 1);
+
+        assertFileSystemAccesses("SHOW STATS FOR (SELECT * FROM test_show_stats_with_filter where age >= 2)",
+                ImmutableMultiset.builder()
+                        .addCopies(new FileOperation(SNAPSHOT, INPUT_FILE_GET_LENGTH), 1)
+                        .addCopies(new FileOperation(SNAPSHOT, INPUT_FILE_NEW_STREAM), 1)
+                        .addCopies(new FileOperation(METADATA, INPUT_FILE_NEW_STREAM), 1)
+                        .addCopies(new FileOperation(MANIFEST, INPUT_FILE_GET_LENGTH), 2)
+                        .addCopies(new FileOperation(MANIFEST, INPUT_FILE_NEW_STREAM), 2)
+                        .build());
+    }
+
+    private void assertFileSystemAccesses(String query, Multiset expectedAccesses)
+            throws Exception
+    {
+        resetCounts();
+        getDistributedQueryRunner().executeWithQueryId(TEST_SESSION, query);
+        assertThat(getOperations()).containsExactlyInAnyOrderElementsOf(expectedAccesses);
+    }
+
+    private static void resetCounts()
+            throws Exception
+    {
+        BEAN_SERVER.invoke(new ObjectName(BEAN_NAME), "reset", null, null);
+    }
+
+    private static Multiset<FileOperation> getOperations()
+            throws Exception
+    {
+        return ((Map<OperationContext, Integer>) BEAN_SERVER.getAttribute(new ObjectName(BEAN_NAME), "OperationCounts"))
+                .entrySet().stream()
+                .map(entry -> entry.getKey())
+                .map(FileOperation::new)
```

Comment:
why are the values being ignored here?

did you mean sth like

```
.flatMap(entry -> nCopies(entry.getValue(), new FileOperation(entry.getKey())).stream())
```

?

Reply:
oh this is definitely what should be there, for some unknown reason I just totally forgot about these values.. Thanks for spotting it otherwise test would be useless